packages/contracts/src/utils/exchange_wrapper.ts

```diff
@@ -3,13 +3,16 @@ import { BigNumber } from '@0xproject/utils';
 import * as _ from 'lodash';
 import * as Web3 from 'web3';
 
-import { ExchangeContract } from '../src/contract_wrappers/generated/exchange';
+import { ExchangeContract } from '../contract_wrappers/generated/exchange';
 
+import { constants } from './constants';
 import { formatters } from './formatters';
+import { LogDecoder } from './log_decoder';
 import { signedOrderUtils } from './signed_order_utils';
 
 export class ExchangeWrapper {
     private _exchange: ExchangeContract;
+    private _logDecoder: LogDecoder = new LogDecoder(constants.TESTRPC_NETWORK_ID);
```

Comment:
Do you expect to instantiate `logDecoder` in multiple places?

Reply:
Probably not, why?

Reply:
Because if so, there would be no need for the `LogDecoder` class. But perhaps it's cleaner this way.